core/ustring.cpp

```diff
@@ -39,6 +39,7 @@
 #include "core/ucaps.h"
 #include "core/variant.h"
 
+#include <limits.h>
```

Comment:
That should likely be `<cstdint>` as per https://en.cppreference.com/w/cpp/types/integer, currently it doesn't build on AppVeyor.

And maybe we should use `INT32_MAX` instead of `INT_MAX`?

Reply:
INT_MAX and INT_MIN doesn't exists in <cstdint>, so INT32_MAX and INT32_MIN are the only solution.